src/IntegratedExperts/BehatScreenshotExtension/Context/ScreenshotContext.php

```diff
@@ -117,16 +119,29 @@ public function iSaveScreenshot()
         $driver = $this->getSession()->getDriver();
 
         if ($driver instanceof Selenium2Driver) {
-            $data = $this->getSession()->getScreenshot();
-            $ext = 'png';
+            // If both png and html are not set assume the default of saving png screen shots.
+            if ($this->png || empty($this->png) && empty($this->html)) {
+                $data = $this->getSession()->getScreenshot();
+                $ext = 'png';
+                if ($data) {
```

Comment:
these checks are used 3 times here - please move them below after the code that identifies the type and data.